### ISSUE TITLE
Get message id for replies to document

### ISSUE THREAD

**Issue by alishahlakhani** [smashah/sulla#216]
Is it possible to get Message id for replies to messages type "document" in the **message.quotedMsg** object? I wish to use that message id and forward the message.

**Steps:**
Steps to reproduce the behaviour:
1. User A asks a question by sending a message.
2. User B replies with a document.
3. User A replies to the (2) with a quoted message.
4. System should be able to get the message id in (2) and forward that document

When I write **message.quotedMsg.id**, it gives me undefined.

I wish to get the message ID for (2) for I could forward this document using it's message id.

**Comment by smashah** [comment]
@alishahlakhani can you provide a screenshot for clarity, please. I'm assuming this would be an issue with any type of file, not just document.

Do you want to get the message-id for 2 from 3? what type of message is 3?

**Comment by smashah** [comment]
@alishahlakhani please try `message.quotedMsgObj.id`

**Comment by JoeDozan24** [comment]
I added a function at the whatsapp.js
```js
Whatsapp.prototype.getMessageById = function (chatId) {
    return __awaiter(this, void 0, void 0, function () {
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4, this.page.evaluate(function (_a) {
                        var chatId = _a.chatId;
                        var x = WAPI.getMessageById(chatId);
                        return x;
                    }, { chatId: chatId })];
                case 1: return [2, _a.sent()];
            }
        });
    });
};
```
and I called it like this

```js
let messageIds = JSON.parse(messageIds); // messageIds is an Array of message Id
let messages = [];
for(let i=0;i<messageIds.length;i++) {
	const msgToForward = await user.getMessageById(messageIds[i]);
	messages.push(msgToForward);
}			
await user.forwardMessages(to, messages);
```
I use the getMessageById only for getting the message obj. Same for Reply but only for 1 id
```js
const msgToReply = await user.getMessageById(messageId);
await user.reply(to, caption, msgToReply);
```

**Comment by alishahlakhani** [comment]
> @alishahlakhani please try `message.quotedMsgObj.id`

it returns me 'TypeError: Cannot read property 'id' of null'. It seems like 'quotedMsgObj/quotedMsg' is null. Also why's there 'quotedMsgObj' and 'quotedMsg'?

**Comment by alishahlakhani** [comment]
> @alishahlakhani can you provide a screenshot for clarity, please. I'm assuming this would be an issue with any type of file, not just document.
> 
> Do you want to get the message-id for 2 from 3? what type of message is 3?

The issue I'm facing is that I am unable to get chat id for reply messages. The id should be there in   'quotedMsgObj/quotedMsg' if I'm not wrong but I don't see it.

**Comment by alishahlakhani** [comment]
@smashah Any idea how to resolve it? I am trying to understand the sulla's underlying code but It's taking time.

**Comment by smashah** [comment]
@alishahlakhani  the reason this happens is because you need to call `quotedMsgObj()` to get all the details of the quoted message, not just the media data. You should now be able to use the following to get the id of the quoted document message:

`message.quotedMsgObject.id`

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i`);
exec(`npm run release-ci`);

var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [1.8.26](https://github.com/smashah/sulla/compare/1.8.25...1.8.26) (2020-03-29)

- Release 1.8.26 [`aeb5b7a`](https://github.com/smashah/sulla/commit/aeb5b7a8827c54992c3aab1f754ff3f8fa457b8f)
- possible fix for #216 [`7316133`](https://github.com/smashah/sulla/commit/731613381b274ac82f86042a5b88c249986d3b2b)
- tested and confirmed fix for #216 [`7b5c293`](https://github.com/smashah/sulla/commit/7b5c293fc0d8cb20471865193c07d34c15585050)

**Comment by alishahlakhani** [comment]
Works like a charm! every time. Thanks @smashah 😃
